matscipy/calculators/manybody/newmb.py

```diff
@@ -641,3 +643,45 @@ def get_hessian(self, atoms, format='sparse', divide_by_masses=False):
         )
 
         return H
+
+class HessianPrecon(Precon):
+    def __init__(self, calc, c_stab=0.01, move_tol=0.1, P=None, old_positions=None):
+        self.calc = calc
+        self.P = P        
+        self.c_stab = c_stab
+        self.move_tol = move_tol
+        self.old_positions = old_positions        
+        
+    def make_precon(self, atoms):
+        if self.P is None or self.old_positions is None:
+            max_move = np.inf
+        else:
+            max_move = np.abs(atoms.positions - self.old_positions).max()
+        if self.P is None or max_move > self.move_tol:
+            print('Recomputing Hessian...')
+            P = self.calc.get_hessian(atoms, format='dense')
```

Comment:
Using `get_hessian` excludes mixin calculators. Better to use `get_property("hessian")`. See `tests/manybody/test_newmb.py` for examples.